src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs

```diff
@@ -562,7 +562,8 @@ private static string InsertSegments(string originalString, IEnumerable<(int ori
 
             if (TestWorkspace.IsWorkspaceElement(expectedText))
             {
-                await VerifyAgainstWorkspaceDefinitionAsync(expectedText, newSolution, workspace.ExportProvider);
+                var newSolutionWithLinkedFiles = await newSolution.WithMergedLinkedFileChangesAsync(oldSolution);
+                await VerifyAgainstWorkspaceDefinitionAsync(expectedText, newSolutionWithLinkedFiles, workspace.ExportProvider);
```

Comment:
Should this be opt-in in some way? I'm worried this will hide bad code which doesn't account for linked files, but the tests don't fail, or make bugs harder to fix by making it harder to make a failing test.

Reply:
i don't think so.  We really want this.  Code actions shouldn't need to update other parts of a linked file.  They just update one, and the workspace does this.  The primary issue here is htat we're not going through the workspace, so we don't get that behavior.

All this code goes away as we move more and more tests over to the roslyn-sdk (which does the right thing and goes through the workspace).